### PR TITLE
Implement filter option for negated labels

### PR DESCRIPTION
**Pull request by zebrapurring**
## What type of PR is this?

- feature

## What this PR does / why we need it:

Adds an item filtering option to negate all the selected labels.

## Which issue(s) this PR fixes:

No issue.

## Testing

Select one or more labels and toggle the new checkbox `Negate selected labels` inside `Options`.

## Release Notes

```release-note
Add filter option for negated labels
```

## Summary by CodeRabbit

- **New Features**
  - Added a checkbox in the UI to toggle label negation in item searches.

- **Improvements**
  - Enhanced search functionality to include an option to negate labels.
  - Updated debugging configurations to use the integrated terminal in VSCode.

- **Chores**
  - Updated .gitignore to exclude `__debug_bin*` directories.
  - Improved date formatting in Docker publish workflows.

### CODE REVIEW

**Comment by katosdev** [comment]
Thanks for the PR @zebrapurring ! 
Requesting approval by @tankerkiller125

File: .gitignore
Comment (katosdev) [inline .gitignore:1]: Confused by this one, not sure that the * is required? Will be honest though, I'm likely missing something silly!

Comment (zebrapurring) [reply .gitignore:1]: I noticed when I launch the backend debug configuration on VSCode it creates the `__debug_bin` file with a random suffix, maybe depends on the Golang version? I'm using `go1.22.4`.

golang_debug

Comment (katosdev) [reply .gitignore:1]: > I noticed when I launch the backend debug configuration on VSCode it creates the `__debug_bin` file with a random suffix, maybe depends on the Golang version? I'm using `go1.22.4`.
> 

Ahh! Good spot, this makes sense.

**Pull request merged**